Source files using `--login` for bash and zsh

oxalica opened the pull request:
Simplify codes and make `zsh` also source `/etc/profile`.
`/etc/profile` contains basic environment variables like `QT_IM_MODULE` on NixOS. Ignoring it will make input method unusable.

oxalica committed:
Source files using `--login` for bash and zsh

marcthe12 commented:
there is Xsession script also which performs the same logic. I suggest to do the necessary changes to that script also

oxalica commented:
@marcthe12 Fixed.

davidedmundson reviewed (approved):
It does indeed seem simpler and appears to work the same

tavianator commented on data/scripts/Xsession:
```diff
@@ -8,26 +8,11 @@
 # Note that the respective logout scripts are not sourced.
```
I think this won't be true any more

oxalica replied:
We `exec` at the end of script, I confirmed logout scripts have no possibility to run when we  `exec`.

tavianator replied:
Oh true, never mind!

oxalica committed:
Check to ensure we are in a login shell

When /bin/sh is linked to a shell same as $SHELL, we should re-exec
it to ensure it is a login shell, to have login scripts sourced.

plfiorini committed:
Merge branch 'develop' into fix/shell-preload

oxalica committed:
Merge branch 'develop' into fix/shell-preload

plfiorini committed:
Merge branch 'develop' into fix/shell-preload

plfiorini committed:
Merge branch 'develop' into fix/shell-preload
